h/search/query.py

```diff
@@ -22,19 +22,25 @@ def wildcard_uri_is_valid(wildcard_uri):
     """
     Return True if uri contains wildcards in appropriate places, return False otherwise.
 
-    Wildcards are not permitted in the scheme or netloc of the uri.
+    *'s are not permitted in the scheme or netloc. ?'s are not permitted in the scheme.
     """
     if "*" not in wildcard_uri and "?" not in wildcard_uri:
         return False
     try:
-        normalized_uri = urlparse.urlparse(wildcard_uri.replace("*", "").replace("?", ""))
-
-        # Remove all parts of the url except the scheme, netloc, and provide a substitute
-        # path value "p" so that uri's that only have a scheme and path are still valid.
-        uri_parts = (normalized_uri.scheme, normalized_uri.netloc, "p", "", "", "")
+        normalized_uri = urlparse.urlparse(wildcard_uri.replace("*", ""))
+        # If the normalized uri has no path component but has a query component.
+        if (not normalized_uri.path and
+                normalized_uri.netloc and
+                normalized_uri.query):
+            uri_parts = (normalized_uri.scheme, normalized_uri.netloc, "", normalized_uri.query, "", "")
```

Comment:
The urlparse result is a `(scheme, netloc, path, params, query, fragment)`, so you are putting the _query_ into the _params_ section and then undoing the swap with the "/;" => "?" change. Was that intended?